### PR TITLE
Update haskell layer documenation.

### PR DESCRIPTION
**Pull request by maxigit**
+ add syntax checking section
+ improve FAQ to solve common syntaxk checking problems.

### CODE REVIEW

**Comment by d12frosted** [comment]
Reincarnation of #5963. 

Just FYI, you can squash commits and force-push them to update existing PR. So all conversations about one PR don't scatter around :) In case you need assistance in this - just let us know.

File: layers/+lang/haskell/README.org [hunk 1]
```diff
@@ -408,3 +465,62 @@ This is intended behavior in =haskell-indentation-mode=. If you want to reset in
...
+Check the [[Flycheck doesn't work]] section.
+** Flycheck doesn't work
+You can check what is wrong with flycheck with the =flycheck-compile= command. 
+This will show you the exact command line used and its output.
+
+If you are using stack, check the [[Flycheck doesn't work with =stack=]] section.
+
+** Flycheck doesn't work with =stack=
+First check flycheck use the correct checker and all the path are properly configured using =flycheck-verify-setup= (~SPC e v~).
+You can force the checker by using =flycheck-set-checker= (~SPC e s~) to =haskell-stack-ghc=.
```
Comment (joehillen) [inline layers/+lang/haskell/README.org]: It's `flycheck-select-checker`

File: layers/+lang/haskell/README.org [hunk 2]
```diff
@@ -408,3 +465,62 @@ This is intended behavior in =haskell-indentation-mode=. If you want to reset in
...
+#!/bin/bash
+cd `stack --project-root`
+stack $*
+#+END_SRC
+
+and set the =flycheck-haskell-stack-ghc-executable= to it.
+
+** haskell-mode commands don't work 
+Some (most of) haskell-mode commands only works when haskell-mode is in interactive mode, .i.e as a interactive session associated to it.
+Load the buffer using ~SPC s b~.
```
Comment (joehillen) [inline layers/+lang/haskell/README.org]: `SPC m s b` or `, s b`

**Comment by TheBB** [comment]
Thanks! Cherry-picked in develop. You can safely delete your branch.

I went to town on the README in https://github.com/syl20bnr/spacemacs/commit/3145e70f72bd7d471f2f55f40d005e8d60000316 to fix various English errors and line length issues. Just FYI! I also referred to the syntax-checking documentation instead of listing every binding over again.

**Comment by d12frosted** [comment]
:+1: nice

P. S. thanks for the update :)

~ d12frosted

**Pull request closed without merging**